cache/src/test/java/io/envoyproxy/controlplane/cache/CacheStatusInfoTest.java

```diff
@@ -0,0 +1,83 @@
+package io.envoyproxy.controlplane.cache;
+
+import static org.assertj.core.api.Assertions.assertThat;
+
+import envoy.api.v2.Discovery.DiscoveryRequest;
+import envoy.api.v2.core.Base.Node;
+import java.util.UUID;
+import java.util.concurrent.ThreadLocalRandom;
+import org.junit.Test;
+
+public class CacheStatusInfoTest {
```

Comment:
can we add one multithreaded read/write test to ensure the consistency across writes/reads by each of the threads?